src/dc_msg.rs

```diff
@@ -1440,125 +1392,13 @@ pub fn dc_update_server_uid(
-                &mut mime_0,
-            );
-            assert_eq!(type_0, Viewtype::Image);
-            assert_eq!(
-                CStr::from_ptr(mime_0 as *const libc::c_char)
-                    .to_str()
-                    .unwrap(),
-                "image/jpeg"
-            );
-            free(mime_0 as *mut libc::c_void);
-
-            dc_msg_guess_msgtype_from_suffix(
-                b"foo/bar-sth.jpeg\x00" as *const u8 as *const libc::c_char,
-                &mut type_0,
-                &mut mime_0,
-            );
-            assert_eq!(type_0, Viewtype::Image);
-            assert_eq!(
-                CStr::from_ptr(mime_0 as *const libc::c_char)
-                    .to_str()
-                    .unwrap(),
-                "image/jpeg"
-            );
-            free(mime_0 as *mut libc::c_void);
-
-            dc_msg_guess_msgtype_from_suffix(
-                b"foo/bar-sth.png\x00" as *const u8 as *const libc::c_char,
-                &mut type_0,
-                &mut mime_0,
-            );
-            assert_eq!(type_0, Viewtype::Image);
-            assert_eq!(
-                CStr::from_ptr(mime_0 as *const libc::c_char)
-                    .to_str()
-                    .unwrap(),
-                "image/png"
-            );
-            free(mime_0 as *mut libc::c_void);
-
-            dc_msg_guess_msgtype_from_suffix(
-                b"foo/bar-sth.webp\x00" as *const u8 as *const libc::c_char,
-                &mut type_0,
-                &mut mime_0,
-            );
-            assert_eq!(type_0, Viewtype::Image);
-            assert_eq!(
-                CStr::from_ptr(mime_0 as *const libc::c_char)
-                    .to_str()
-                    .unwrap(),
-                "image/webp"
-            );
-            free(mime_0 as *mut libc::c_void);
-
-            dc_msg_guess_msgtype_from_suffix(
-                b"foo/bar-sth.gif\x00" as *const u8 as *const libc::c_char,
-                &mut type_0,
-                &mut mime_0,
-            );
-            assert_eq!(type_0, Viewtype::Gif);
-            assert_eq!(
-                CStr::from_ptr(mime_0 as *const libc::c_char)
-                    .to_str()
-                    .unwrap(),
-                "image/gif"
-            );
-            free(mime_0 as *mut libc::c_void);
-
-            dc_msg_guess_msgtype_from_suffix(
-                b"foo/bar-sth.vcf\x00" as *const u8 as *const libc::c_char,
-                &mut type_0,
-                &mut mime_0,
-            );
-            assert_eq!(type_0, Viewtype::File);
-            assert_eq!(
-                CStr::from_ptr(mime_0 as *const libc::c_char)
-                    .to_str()
-                    .unwrap(),
-                "text/vcard"
-            );
-            free(mime_0 as *mut libc::c_void);
-        }
+        assert_eq!(
+            dc_msg_guess_msgtype_from_suffix(Path::new("foo/bar-sth.mp3")),
+            Some((Viewtype::Audio, "audio/mpeg"))
+        );
```

Comment:
forgot to ask: is there a reason why you stroke all the tests above?

Reply:
They are essentially duplicate the code. I do not see point it testing `map.get(key)` code.

I left one to check that my understanding of `extension` is same as `Rust`'s.